controllers/machine_controller.go

```diff
@@ -363,6 +367,21 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	return ctrl.Result{}, nil
 }
 
+func isNodeDraintimeoutOver(machine *clusterv1.Machine) bool {
```

Comment:
```suggestion
func isNodeDrainAllowed(machine *clusterv1.Machine) bool {
```

To make it a little bit more generic, in case in the future we might want to check more than just a timeout. We can also add the check for the annotation `if _, exists := m.ObjectMeta.Annotations[clusterv1.ExcludeNodeDrainingAnnotation]`